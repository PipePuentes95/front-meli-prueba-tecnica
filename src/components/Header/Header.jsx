import React, { useState } from 'react'
import { useNavigate } from 'react-router-dom';

import './Header.scss';
import logo from '../../assets/logo-meli.png';
import icon_search from '../../assets/icon-search.png';

const Header = () => {
  const navigate = useNavigate();
  const [searchValue, setSearchValue] = useState('');

  const handlerInputSearch = (e) => {
    e.preventDefault();
    setSearchValue(e.target.value);
  };

  const handlerSearch = (e) => {
    e.preventDefault();
    return navigate(`/items?q=${ searchValue }`);
  };

  const handlerGoToHome = (e) => {
    return navigate('/');
  }

  return (
    <header className='header-wrapper'>
      <div className="container">
        <div className="header">
        
          <button className='header-btn-logo' onClick={ handlerGoToHome } data-testid="go-to-home">
            <img className='header-img' src={ logo } alt="logo mercado libre" width="53" height="36" />
          </button>
          
          <form className='header-search' onSubmit={ handlerSearch } data-testid="go-items">
            <input
              className='search-bar'
              placeholder='Nunca dejes de buscar'
              type="text"
              onChange={ handlerInputSearch }
              data-testid="search-bar" />
            <button className='search-btn' type='submit'>
              <img src={ icon_search } alt="icon search" width="18" height="18" />
            </button>
          </form>
        </div>
      </div>
    </header>
  );
};

export default Header;
