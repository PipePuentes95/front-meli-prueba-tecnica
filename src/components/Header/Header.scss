@import '../../styles/variables';

.header-wrapper {
  background-color: $yellow-1;

  .header {
    @media(max-width: 764px) {
      width: 100%;
    }

    width: 80%;

    display: flex;
    align-items: center;

    margin: 0 auto;
    padding: 10px 0;

    &-img { margin-right: 24px; }

    &-search {
      position: relative;
      width: 100%;

      .search-bar {
        width: 100%;
        height: 40px;
        padding: 10px 14px;

        font-size: 18px;
        color: $grey-2;

        border: none;
        border-radius: 4px;

        &::placeholder { color: $grey-3; }
      }

      .search-btn {
        position: absolute;
        right: 0;
        top: 0;

        display: flex;
        justify-content: center;
        align-items: center;

        width: 45px;
        height: 40px;

        border: none;
        border-radius: 0 4px 4px 0;

        cursor: pointer;
      }
    }
  }
}