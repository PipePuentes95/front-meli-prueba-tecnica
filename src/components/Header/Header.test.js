import { render, fireEvent, screen } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import Header from './Header';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => (jest.fn())
})); 

it('should take a snapshot', () => {
  const pathname = createMemoryHistory();
  const { asFragment } = render (
    <Router location={ pathname.location }>
      <Header />
    </Router>
  )

  expect(asFragment(<Header />)).toMatchSnapshot();
});

it('should detect value of input', () => {
  const pathname = createMemoryHistory();

  render (
    <Router location={ pathname.location }>
      <Header />
    </Router>
  )

  fireEvent.change(screen.getByTestId('search-bar'), { target: { value: "adidas"} });
  expect(screen.getByTestId('search-bar')).toHaveValue("adidas")
});

it('should go to home', () => {
  const history = createMemoryHistory();

  render (
    <Router location={ history.location }>
      <Header />
    </Router>
  )

  fireEvent.click(screen.getByTestId('go-to-home'));
  expect(history.location.pathname).toBe('/');
});
