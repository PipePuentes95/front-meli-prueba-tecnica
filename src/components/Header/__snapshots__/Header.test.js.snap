// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="header-wrapper"
  >
    <div
      class="container"
    >
      <div
        class="header"
      >
        <button
          class="header-btn-logo"
          data-testid="go-to-home"
        >
          <img
            alt="logo mercado libre"
            class="header-img"
            height="36"
            src="logo-meli.png"
            width="53"
          />
        </button>
        <form
          class="header-search"
          data-testid="go-items"
        >
          <input
            class="search-bar"
            data-testid="search-bar"
            placeholder="Nunca dejes de buscar"
            type="text"
          />
          <button
            class="search-btn"
            type="submit"
          >
            <img
              alt="icon search"
              height="18"
              src="icon-search.png"
              width="18"
            />
          </button>
        </form>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
